### PR TITLE
fix: server reinstall

### PR DESCRIPTION
**Pull request by ynhummel**
#### What does this PR do?
This PR aims to avoid cases where users could unintentionally trigger a server reinstall.

We verify if a reinstall operation will take place at runtime inside the Update function, which makes it hard to warn the user before the reinstall occurs, and Terraform only reports an in-place update.

#### Description of Task to be completed?
Create an `allow_reinstall` variable that will be used to let Terraform trigger the reinstall. This variable will be false by default.
Make general adjustments to the reinstall process to avoid errors.

#### How should this be manually tested?
to test the provider manually, you can follow [Notion - Developing the Terraform Provider Locally](https://www.notion.so/Developing-the-Terraform-Provider-Locally-6682fe3df39d4003b519301e630f51ef?pvs=27&qid=1%3A8c80df71-563c-4c6a-905d-be4a225ec805%3A0)

To verify:
- Creating a server with a tag will trigger a subsequent update. This update should not reinstall the server even if the triggering attributes (ssh_keys, user_data, raid, or ipxe_url) are given.
- Changing the allow_reinstall variable itself should not trigger a reinstall.

### CODE REVIEW

File: latitudesh/resource_server.go [hunk 1]
```diff
@@ -324,9 +329,17 @@ func serverReinstall(c *api.Client, serverID string, ctx context.Context, d *sch
...
 
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
-	return nil
+
+	diags = append(diags, diag.Diagnostic{
+		Severity: diag.Warning,
+		Summary:  "Your Server is being reinstalled",
```
Comment (esoubihe) [inline latitudesh/resource_server.go:343]:
```suggestion
		Summary:  "Your server is being reinstalled",
```

File: latitudesh/resource_server.go [hunk 2]
```diff
@@ -324,9 +329,17 @@ func serverReinstall(c *api.Client, serverID string, ctx context.Context, d *sch
...
+		Detail: "[WARN] The changes made to your server resource require a reinstallation. " +
```
Comment (esoubihe) [inline latitudesh/resource_server.go:344]:
```suggestion
		Detail: "[WARN] The changes made to the server resource will trigger a reinstallation. All disks will be erased." +
```

**Comment by esoubihe** [comment]
> * Changing the allow_reinstall variable itself should not trigger a reinstall.

What do you mean by this?

**Comment by ynhummel** [comment]
> > * Changing the allow_reinstall variable itself should not trigger a reinstall.
> 
> What do you mean by this?

When I was first testing, there was a bug where modifying the allow_reinstall variable would trigger a reinstallation even if nothing else changed. Now this isn't occurring anymore, but I thought it was a good idea to warn that this once happened.

**Review by leonardorisch — APPROVED** [review]

**Pull request merged**